CoreCLR tests failed with Mono interpreter: Interop/NativeLibrary/API/NativeLibraryTests

**Repro:** 
1. cd src/mono/netcore
2. export MONO_ENV_OPTIONS=--interpreter
3. make run-tests-coreclr CoreClrTest="bash ../../artifacts/tests/coreclr/OSX.x64.Debug/Interop/NativeLibrary/API/NativeLibraryTests/NativeLibraryTests.sh"

**Log:**
```
  BEGIN EXECUTION
  /Users/fanyang/Documents/work/dotnet_test/runtime/artifacts/tests/coreclr/OSX.x64.Debug/Tests/Core_Root/corerun NativeLibraryTests.dll ''
  Unhandled exception System.DllNotFoundException: /Users/fanyang/Documents/work/dotnet_test/runtime/artifacts/tests/coreclr/OSX.x64.Debug/Interop/NativeLibrary/API/NativeLibraryTests/notfound
     at System.String.Format(String format, Object arg0, Object arg1) in /Users/fanyang/Documents/work/dotnet_test/runtime/src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs:line 475
     at NativeLibraryTest.EXPECT(TestResult actualValue, TestResult expectedValue) in /Users/fanyang/Documents/work/dotnet_test/runtime/src/coreclr/tests/src/Interop/NativeLibrary/API/NativeLibraryTests.cs:line 206
     at NativeLibraryTest.Main() in /Users/fanyang/Documents/work/dotnet_test/runtime/src/coreclr/tests/src/Interop/NativeLibrary/API/NativeLibraryTests.cs:line 78
  Expected: 100
  Actual: 156
  END EXECUTION - FAILED
```

Simplified repro:

`using System;
using System.IO;
using System.Reflection;
using System.Runtime.InteropServices;

namespace HelloWorld
{
    class Program
    {
        static void Main(string[] args)
        {
	    Assembly assembly = System.Reflection.Assembly.GetExecutingAssembly();
	    string assemblyDir = Path.GetDirectoryName(assembly.Location);
	    string libName = Path.Combine(assemblyDir, "notfound");
	    IntPtr handle = IntPtr.Zero;
	    try
	    {
		    handle = NativeLibrary.Load(libName, assembly, null);
	    }
	    catch (Exception e)
	    {
		    Console.WriteLine("{0} Exception caught.", e);
	    }
	    NativeLibrary.Free(handle);
        }
    }
}`

Expected:
`System.DllNotFoundException: /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/bin/notfound
   at System.Runtime.InteropServices.NativeLibrary.LoadLibraryByName(String libraryName, Assembly assembly, Nullable`1 searchPath, Boolean throwOnError) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.Mono.cs:line 13
   at System.Runtime.InteropServices.NativeLibrary.Load(String libraryName, Assembly assembly, Nullable`1 searchPath) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs:line 94
   at HelloWorld.Program.Main(String[] args) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/Program.cs:line 18 Exception caught.`

Interpreter mode without fix:
`Unhandled Exception:
System.DllNotFoundException: /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/bin/notfound
   at System.Runtime.InteropServices.NativeLibrary.Free(IntPtr handle) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs:line 148
   at HelloWorld.Program.Main(String[] args) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/Program.cs:line 24
[ERROR] FATAL UNHANDLED EXCEPTION: System.DllNotFoundException: /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/bin/notfound
   at System.Runtime.InteropServices.NativeLibrary.Free(IntPtr handle) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs:line 148
   at HelloWorld.Program.Main(String[] args) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/Program.cs:line 24`

Follow-up to my previous comment:
**Expected:**

> System.DllNotFoundException: /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/bin/notfound
>    at System.Runtime.InteropServices.NativeLibrary.LoadLibraryByName(String libraryName, Assembly assembly, Nullable`1 searchPath, Boolean throwOnError) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.Mono.cs:line 13
>    at System.Runtime.InteropServices.NativeLibrary.Load(String libraryName, Assembly assembly, Nullable`1 searchPath) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs:line 94
>    at HelloWorld.Program.Main(String[] args) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/Program.cs:line 18 Exception caught.

**Interpreter mode without fix:**
Unhandled Exception:
System.DllNotFoundException: /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/bin/notfound
   at System.Runtime.InteropServices.NativeLibrary.Free(IntPtr handle) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs:line 148
   at HelloWorld.Program.Main(String[] args) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/Program.cs:line 24
[ERROR] FATAL UNHANDLED EXCEPTION: System.DllNotFoundException: /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/bin/notfound
   at System.Runtime.InteropServices.NativeLibrary.Free(IntPtr handle) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs:line 148
   at HelloWorld.Program.Main(String[] args) in /Users/fanyang/Documents/work/dotnet_latest/runtime/src/mono/netcore/sample/HelloWorld/Program.cs:line 24

--- a/src/coreclr/tests/issues.targets
+++ b/src/coreclr/tests/issues.targets
@@ -2339,9 +2339,6 @@
         <ExcludeList Include = "$(XunitTestBinBase)/JIT/IL_Conformance/Old/Conformance_Base/neg_r4/**">
             <Issue>needs triage</Issue>
         </ExcludeList>
-        <ExcludeList Include = "$(XunitTestBinBase)/Interop/NativeLibrary/API/NativeLibraryTests/**">
-            <Issue>needs triage</Issue>
-        </ExcludeList>
         <ExcludeList Include = "$(XunitTestBinBase)/JIT/CodeGenBringUpTests/Localloc_d/**">
             <Issue>needs triage</Issue>
         </ExcludeList>

--- a/src/mono/mono/mini/interp/interp.c
+++ b/src/mono/mono/mini/interp/interp.c
@@ -3847,6 +3847,7 @@ main_loop:
 			frame->state.ip = ip + 6;
 			ves_pinvoke_method (csignature, (MonoFuncV)code, context, frame, &retval, save_last_error, cache, sp);
 
+			EXCEPTION_CHECKPOINT_GC_UNSAFE;
 			CHECK_RESUME_STATE (context);
 
 			if (csignature->ret->type != MONO_TYPE_VOID) {

